src/core/common/instance.cpp

```diff
@@ -125,15 +125,18 @@ Instance::Instance(void)
 
 #if !OPENTHREAD_CONFIG_MULTIPLE_INSTANCE_ENABLE
 
-Instance &Instance::InitSingle(void)
+Instance &Instance::InitSingle(bool aDryRun)
 {
     Instance *instance = &Get();
 
     VerifyOrExit(!instance->mIsInitialized);
 
     instance = new (&gInstanceRaw) Instance();
 
-    instance->AfterInit();
+    if (!aDryRun)
```

Comment:
It's not logically clear why we don't run `AfterInit()` in `dry-run` mode.

I think we can keep the OT code unchanged (i.e. without adding `aDryRun` argument to `otInstanceInitSingle` API), but implement `dry-run` mode totally in Posix platform.

Thoughts?

Reply:
Since you already added a flag in platform config, we may give settings module the access to the config and do not create files when in dry run mode.

Reply:
We can avoid creating files directly in posix `otPlatSettingsInit` without changing how core `settings` works.

I think it's better not to pollute OT core with a `aDryRun` argument.

Reply:
Thanks! Sounds good to me. But I still have one question: `otPlatSettingsInit` currently accepts a parameter of `otInstance`. To let it know if it's in `DryRun` mode, we still need to add another parameter to indicate that. Thoughts? @simonlingoogle @bukepo

Reply:
I think we can get all platform config in posix platform `otSysInit`. So, we should be able to save any information useful for other parts of posix platform to access.

Reply:
Do you mean we save it into a static variable in `system.cpp` and in `otPlatSettingsInit` we call another posix API to access the variable in `system.cpp`? This is the only way I can think of. @simonlingoogle

Reply:
> Do you mean we save it into a static variable in `system.cpp` and in `otPlatSettingsInit` we call another posix API to access the variable in `system.cpp`? This is the only way I can think of. @simonlingoogle

I think it should work.

Reply:
Updated.